### PR TITLE
 Add Kdocs on `ChatClient.Builder` methods, deprecate `cdnUrl`

### PR DESCRIPTION
**Pull request by zsmb13**
### 🎯 Goal

Make `ChatClient.Builder` methods more obvious.

More info on `cdnUrl` deprecation: [Slack discussion](https://getstream.slack.com/archives/CE5N802GP/p1633338690250200)

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [ ] ~Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -1574,21 +1580,61 @@ public class ChatClient internal constructor(
...
             this.notificationsHandler = notificationsHandler
             return this
         }
 
+        /**
+         * Sets a custom file uploader implementation that will be used by the client
+         * to upload files and images.
+         *
+         * The default implementation uses Stream's own CDN to store these files,
+         * which has a 20 MB upload size limit.
```
Comment (bychkovdmitry) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1632]: We need to review this ASAP, I'm almost sure clients will still not be able to send files via UI components as this limit is hardcoded there.

**Review by bychkovdmitry — APPROVED** [review]
Great job!

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt (hunk 1 above)
Comment (zsmb13) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1632]: I believe this is handled with the [`streamUiAttachmentMaxFileSizeMb`](https://github.com/GetStream/stream-chat-android/blob/6e6354373b6011dedc4b152f4aac04bd68fe05f5/stream-chat-android-ui-components/src/main/res/values/attrs_message_input_view.xml#L94) attribute of `MessageInputView`. That value gets passed into `MessageInputFieldView`, which seems to use it correctly.

Am I missing something else? Should we also handle this somewhere else in the code?

Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1632]: Sorry, you are right. Just always get confused about this line in our adapter:
```
internal var attachmentMaxFileSize: Long = AttachmentConstants.MAX_UPLOAD_FILE_SIZE
```

Comment (zsmb13) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1632]: Ah yes, that's just by default, but it should be updated correctly when it's overridden.

**Pull request merged**